python/lsst/daf/butler/datastore/generic_base.py

```diff
@@ -32,136 +32,35 @@
-            already exists).
-        """
-        raise NotImplementedError()
-
-    @abstractmethod
-    def getStoredItemsInfo(self, ref: DatasetRef) -> Sequence[Any]:
-        """Retrieve information associated with files stored in this
-        `Datastore` associated with this dataset ref.
-
-        Parameters
-        ----------
-        ref : `DatasetRef`
-            The dataset that is to be queried.
-
-        Returns
-        -------
-        items : `list` [`StoredDatastoreItemInfo`]
-            Stored information about the files and associated formatters
-            associated with this dataset. Only one file will be returned
-            if the dataset has not been disassembled. Can return an empty
-            list if no matching datasets can be found.
-        """
-        raise NotImplementedError()
-
-    @abstractmethod
-    def removeStoredItemInfo(self, ref: DatasetRef) -> None:
-        """Remove information about the file associated with this dataset.
-
-        Parameters
-        ----------
-        ref : `DatasetRef`
-            The dataset that has been removed.
-        """
-        raise NotImplementedError()
-
-    def _register_datasets(
-        self,
-        refsAndInfos: Iterable[tuple[DatasetRef, StoredDatastoreItemInfo]],
-        insert_mode: DatabaseInsertMode = DatabaseInsertMode.INSERT,
-    ) -> None:
-        """Update registry to indicate that one or more datasets have been
-        stored.
-
-        Parameters
-        ----------
-        refsAndInfos : sequence `tuple` [`DatasetRef`,
-                                         `StoredDatastoreItemInfo`]
-            Datasets to register and the internal datastore metadata associated
-            with them.
-        insert_mode : `str`, optional
-            Indicate whether the new records should be new ("insert", default),
-            or allowed to exists ("ensure") or be replaced if already present
-            ("replace").
-        """
-        expandedRefs: list[DatasetRef] = []
-        expandedItemInfos = []
-
-        for ref, itemInfo in refsAndInfos:
-            expandedRefs.append(ref)
-            expandedItemInfos.append(itemInfo)
-
-        # Dataset location only cares about registry ID so if we have
-        # disassembled in datastore we have to deduplicate. Since they
-        # will have different datasetTypes we can't use a set
-        registryRefs = {r.id: r for r in expandedRefs}
-        if insert_mode == DatabaseInsertMode.INSERT:
-            self.bridge.insert(registryRefs.values())
-        else:
-            # There are only two columns and all that matters is the
-            # dataset ID.
-            self.bridge.ensure(registryRefs.values())
-        self.addStoredItemInfo(expandedRefs, expandedItemInfos, insert_mode=insert_mode)
-
     def _post_process_get(
         self,
-        inMemoryDataset: Any,
+        inMemoryDataset: object,
```

Comment:
Any particular reason for `object` vs. `Any` here?

Reply:
This is mostly personal preference, I think mypy can catch more potential issues with `object` as `Any` disables all checks. Some docstrings already specified `object` for parameters, this is just restoring documented behavior 🙂

Reply:
`object` is fine.  I was mostly curious for my own education - I knew mypy was a bit more permissive with `Any` but I didn't have any concrete examples of how it was different in my mind.

Reply:
It's probably mostly about accidental corner cases, but here is a trivial example to show differences:
```
from typing import Any

class Butler():
    def get_obj(self) -> object:
        return 42

    def get_any(self) -> Any:
        return 42

b = Butler()

x = [42]
x = b.get_obj()  # will be mypy error
x = b.get_any()  # mypy is OK
l = len(x)  # mypy is still OK, as x is Any here

y = b.get_obj()
y.something()  # mypy error

z = b.get_any()
z.something()  # mypy is OK
```

Reply:
Ah, interesting; `object` is _much_ less permissive than I thought it seems to be an object you can't do anything with (except basic stuff like equality tests), while `Any` permits anything (including implicit downcasts).